docs/sources/next/using-k6-browser/recommended-practices/sleep-and-wait.md

```diff
@@ -0,0 +1,263 @@
+---
+title: 'sleep and wait*'
+description: 'A guide on how to work with sleep and wait*.'
+weight: 04
+---
+
+We will discuss the differences between `sleep` in `k6` and the various `wait*` prepended methods that are available in `k6/browser`. By the end of this page, you should be able to successfully use the correct API where necessary.
+
+{{< admonition type="note" >}}
+
+To avoid flaky tests, avoid working with `sleep` or `page.waitForTimeout` to wait for state changes with an element (such as waiting for elements to appear/disappear or navigating to a different page).
+
+They are, however, useful for mimicking a real user by slowing the test down between API calls, or to debug your test.
```

Comment:
I'd put this line in the first paragraph to avoid confusion. Otherwise, users might think they can still use those and unintentionally face unexpected behavior. Maybe:

> While using the `sleep` function to wait for element state changes can be helpful, it's best to avoid it to prevent unreliable tests. Instead, it's better to use `wait*` prepended methods for reliable tests.

Reply:
I've updated it in 7dd3265c2211b6be2c4fab4c040a3ecef4f1d8c4. WDYT?